Adding Test and Prepare to PR upstream project

systemizer opened the pull request:
Writekey can now be added by specifying an environment variable.

Added a quick test that tests the BuildConfig and a basic Batch test.

Updating the manifest to use updated URI: https://github.com/honeycombio/kubernetes-manifests/pull/1

EDIT: fwiw, this repo seemed to be working before I started working on it. LMK if you're looking for something specific that needs improvement.

systemizer committed:
honeycomb should grab write key from environ

systemizer committed:
Add honeycomb common test

systemizer committed:
Tests are now passing

systemizer committed:
Tests are now passing

systemizer committed:
uncomment vet. this is an upstream issue

systemizer committed:
uncomment vet. this is an upstream issue

emfree commented on common/honeycomb/honeycomb.go:
```diff
@@ -1,13 +1,28 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
```
hmmm not sure we need to add these copyright headers

systemizer replied:
the linting on the project required these, which is why i added them. will probably be required for an upstream merge

emfree replied:
ah okay!

emfree reviewed (commented):
Nice!! I suspect that to get this upstreamed, we'll want to also add something to docs/sink-configuration.md describing how to configure this. And maybe also tests for `events/honeycomb/driver` / `metrics/honeycomb/driver`. But not sure.

systemizer committed:
Merge branch 'systemizer/honeycombsink' of github.com:systemizer/heapster into systemizer/honeycombsink

systemizer commented:
Cool. working on adding docs and tests

systemizer committed:
Adding confugration instructions for honeycomb

systemizer committed:
Adding honeycomb test for metrics

systemizer committed:
Merge branch 'systemizer/honeycombsink' of github.com:systemizer/heapster into systemizer/honeycombsink

systemizer committed:
update honeycomb Client to be interface to make testing easier. add metrics tests

systemizer committed:
the event test found a bug. wooo. fixed and added test

systemizer commented:
@emfree This should do it. added docs, tests (even found a bug with the eventer test!). To get tests to work, I changed Client into an Interface and converted the struct to HoneycombClient. tests pass. also tested locally on minikube. lmk if you have questions

emfree commented on events/sinks/honeycomb/driver.go:
```diff
@@ -53,11 +67,11 @@ func (sink *honeycombSink) ExportEvents(eventBatch *event_core.EventBatch) {
-			glog.Warningf("Failed to send event: %v", err)
-			return
-		}
+	}
```
oof, triple facepalm

emfree commented on common/honeycomb/dummy_honeycomb.go:
```diff
@@ -0,0 +1,42 @@
+}
+
+type FakeHoneycombClient struct {
+	BatchPoints []BatchPointsSavedToHoneycomb
```
Nit: is it necessary to define the `BatchPointsSavedToHoneycomb` type? Could we simply use `BatchPoints[]*BatchPoint` here instead?

emfree commented on common/honeycomb/dummy_honeycomb.go:
```diff
@@ -0,0 +1,42 @@
+	return nil
+}
+
+var FakeClient = NewFakeHoneycombClient()
```
Hmm... package-level globals are generally suspect. It'd be better to call `NewFakeHoneycombClient()` where this object is used, in the `NewFakeClient()` functions. Otherwise, someone may add more tests down the road and be confused as to why state carries over between tests.

emfree reviewed (commented):
Nice! This is looking great. Couple comments inline on testing style, but not blockers IMO.

systemizer committed:
fixes per eben PR: no global. use batchpoints
